lib/internal/file.js

```diff
@@ -21,19 +23,28 @@ const {
 const {
   codes: {
     ERR_MISSING_ARGS,
+    ERR_INVALID_THIS,
   },
 } = require('internal/errors');
 
 const {
   inspect,
 } = require('internal/util/inspect');
 
-class File extends Blob {
-  /** @type {string} */
-  #name;
+const {
+  makeTransferable,
+  kClone,
+  kDeserialize,
+} = require('internal/worker/js_transferable');
 
-  /** @type {number} */
-  #lastModified;
+const kState = Symbol('state');
+
+function isFile(object) {
+  return object?.[kState] !== undefined;
+}
+
+class File extends Blob {
+  [kState] = { __proto__: null };
```

Comment:
This would likely be faster not going through dictionary mode (e.g. extract t to a FileState class)